docs/site/submitting_a_pr.md

```diff
@@ -0,0 +1,629 @@
+
+Click on the `down arrow` on the right-hand side of the `master` branch button.
+
+![submit_pr_create_feature_branch_1.png](./imgs/submit_pr_create_feature_branch_1.png)
+
+This brings up a small dialog for finding or creating a branch.
+
+![submit_pr_create_feature_branch_2.png](./imgs/submit_pr_create_feature_branch_2.png)
+
+Enter the `name` of your new `feature branch` in the search field. Because the branch `emonddr-doc-changes` doesn't yet exist, a menu item to create it appears below the search field. Click on the `Create branch` link.
+
+![submit_pr_create_feature_branch_3.png](./imgs/submit_pr_create_feature_branch_3.png)
+
+Your feature branch is created.
+
+![submit_pr_create_feature_branch_4.png](./imgs/submit_pr_create_feature_branch_4.png)
+
+
+### 4. Clone a local copy of your repository
+
+Click on the `Clone or download` button
+
+![submit_pr_clone_1.png](./imgs/submit_pr_clone_1.png)
+
+This brings up tiny dialog with different choices : Clone with SSH, Use HTTPS, Open in Desktop, or Download Zip).
+
+In my case, I will leave it as `Clone with SSH`, and click on the `copy to clipboard` icon on the right side of the repository url (we are going to paste this value in a terminal window soon).
+
+![submit_pr_clone_2.png](./imgs/submit_pr_clone_2.png)
+
+
+Open a terminal window, and navigate to the directory where you want to clone the repository. In my case, this is `/Users/dremond/git` .
+
+(For all the terminal commands mentioned below, be sure to press the `ENTER` key.)
+
+![submit_pr_clone_3.png](./imgs/submit_pr_clone_3.png)
+
+Type `git clone ` and `Ctrl+V` to paste the github repository url currently in memory.
+
+You should have
+
+```
+git clone git@github.com:{your user id}/{repository name}.git
+```
+
+In my case it is
+
+```
+git clone git@github.com:emonddr/loopback-next.git
+```
+Wait until the cloning process completes.
+
+![submit_pr_clone_4.png](./imgs/submit_pr_clone_4.png)
+
+### 5. Navigate to your repository directory
+
+Type
+
+```
+cd loopback-next
+```
+
+![submit_pr_clone_5.png](./imgs/submit_pr_clone_5.png)
+
+Type
+
+```
+git status
+```
+
+and you will see that you are currently on the `master` branch.
+
+![submit_pr_clone_6.png](./imgs/submit_pr_clone_6.png)
+
+You can also see this by typing
+
+```
+git branch -a
+```
+
+![submit_pr_clone_7.png](./imgs/submit_pr_clone_7.png)
+
+Type `q` to quit viewing the list of branches.
+
+### 6. Checkout your feature branch
+
+You want to start making changes in your feature branch and not your `master` branch, so switch to your `feature` branch.
+
+Type
+
+```
+git checkout emonddr-doc-changes
+```
+
+
+![submit_pr_clone_8.png](./imgs/submit_pr_clone_8.png)
+
+You have successfully switched to your feature branch, and you can also confirm this by typing
+
+```
+git branch -a
+```
+
+![submit_pr_clone_9.png](./imgs/submit_pr_clone_9.png)
+
+Type `q` to quit viewing the list of branches.
+
+
+### 7. Set up remote tracking
```

Comment:
It is only required if the PR needs to be rebased against the official master. We should move this section to advanced steps.

Reply:
@raymondfeng , while I worked on this tutorial with a fake PR request in my forked repo, I had to rebase 2 or 3 times. I agree that it is an advanced topic, but my tutorial doesn't break itself into `basic` and `advanced` sections...it lists all detailed steps in one long list.  It is a thorough tutorial for beginners/intermediates.  I do agree with you ( from another comment) that we should offer 2 choices for a user to follow.  1) Steps for an advanced user (with barely any content..just an outline), and 2) Steps for a beginner/intermediate (with the detailed steps and pictures as I have been providing).